### DISCUSSION TITLE
Using floating-ui with canvas in React

### DISCUSSION THREAD

**Discussion by erquhart** [floating-ui/floating-ui#2217] [Q&A] [answered]
I see that the react hooks like `useHover()` require a proper DOM element, and that `Platform` is available to use floating-ui in non-DOM contexts like canvas. But if I'm using canvas within a React app (which will likely be the prevailing case for canvas), can the hooks be adapted to work with Platform somehow? Just looking to be pointed in the right direction and avoid reinventing things if possible.

**Comment by atomiks** [comment, marked as answer]
The React package is DOM-only, but the positioning does support any platform.  This is assuming your floating element is also canvas-based; if it's a DOM element, you can use a virtual element that matches the canvas reference's location on the screen to position it. None of the interaction hooks will work regardless though.

**Reply by erquhart** [reply]
Got it 👍

**Reply by vaynevayne** [reply]
virtual element What do you mean? @atomiks

**Reply by atomiks** [reply]
Virtual elements can represent an area in the viewport: https://floating-ui.com/docs/virtual-elements
